### ISSUE TITLE
allow yt-dlp to be used instead of youtube-dl

### ISSUE THREAD

**Issue by boredness** [stax76/mpv.net#330]
**Is your feature request related to a problem? Please describe.**
youtube-dl has recently had bandwidth throttling issues with youtube and yt-dlp has resolved that

**Describe the solution you'd like**
add this commit added to mpv recently which allows you to use yt-dlp
https://github.com/mpv-player/mpv/commit/93f84b514a741c2001e6603d0304cd4a6ab63c73

**Comment by hooke007** [comment]
So you just need to download the yt-dlp

**Comment by Daktyl198** [comment]
I've tried downloading yt-dlp.exe and putting it in the same folder as mpvnet.exe, but the bandwidth problems persist. Where should I be placing youtube-dl or yt-dlp for libmpv to pick up on it?

For the record, I've also tried manually pointing to the file using ytdl-path in mpv.conf to no avail.

**Comment by hooke007** [comment]
what's your libmpv version?

**Comment by Daktyl198** [comment]
0.33.0-107

Is there a way to update libmpv separately from mpv.net? I've used "check for updates" on mpv.net and it says I'm up to date.

**Comment by hooke007** [comment]
https://sourceforge.net/projects/mpv-player-windows/files/libmpv/

**Comment by stax76** [comment]
I had tried yt-dlp, and it did not help with my download speed either.

I use the Google Chrome extension [External Application Button](https://chrome.google.com/webstore/detail/external-application-butt/bifmfjgpgndemajpeeoiopbeilbaifdo?hl=en) to download all my YouTube videos, since I switched to Windows 11 every download is slow, before in Windows 10 it was maybe one out of five, to get around this I use the Watch Later feature in YouTube to bookmark wanted videos, and then do mass parallel downloads every other day, often up to 30 videos in parallel.

My command line in External Application Button:

`wt -- pwsh -NoLogo -Command "cd C:\some-folder1\; youtube-dl --ignore-errors --download-archive 'C:\some-folder2\some-file.txt' --output 'D:\some-folder3\%(channel)s - %(title)s.%(ext)s' ('[HREF]' -replace '&list=.+','')"`

**Comment by Daktyl198** [comment]
yt-dlp solved the YouTube bandwidth speeds for me (I just didn't want to spam the issue with a "thank you" comment).

I had already removed youtube-dl from my path and added yt-dlp in it's place. After that, simply upgrading to the latest build of libmpv as hooke007 suggested made it work perfectly. I tested on a 4k video and had 0 download speed issues.

**Comment by SatL1Ng** [comment]
> I had tried yt-dlp, and it did not help with my download speed either.
I downloaded both. youtube-dl and non-vds.

I threw it into the root folder: youtube-dl.exe and the video caching speed was slow. Then I renamed yt-dlp.exe in youtube-dl.exe and also put it in the root folder. Caching speed has become fast. I do not know exactly what the problem is (may be region or simething else), but it just works. So I ask you to add support yt-dlp.exe.
